src/cgroups/v2/freezer.rs

```diff
@@ -0,0 +1,193 @@
+use anyhow::{bail, Result};
+use std::{
+    fs::OpenOptions,
+    io::{BufRead, BufReader, Read, Seek, SeekFrom, Write},
+    path::Path,
+    str, thread,
+    time::Duration,
+};
+
+use oci_spec::{FreezerState, LinuxResources};
+
+use super::controller::Controller;
+
+const CGROUP_FREEZE: &str = "cgroup.freeze";
+const CGROUP_EVENTS: &str = "cgroup.events";
+
+pub struct Freezer {}
+
+impl Controller for Freezer {
+    fn apply(linux_resources: &LinuxResources, cgroup_path: &Path) -> Result<()> {
+        if let Some(freezer_state) = linux_resources.freezer {
+            Self::apply(freezer_state, cgroup_path)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl Freezer {
+    fn apply(freezer_state: FreezerState, path: &Path) -> Result<()> {
+        let state_str = match freezer_state {
+            FreezerState::Undefined => return Ok(()),
+            FreezerState::Frozen => "1",
+            FreezerState::Thawed => "0",
+        };
+
+        match OpenOptions::new()
+            .create(false)
+            .write(true)
+            .open(path.join(CGROUP_FREEZE))
+        {
+            Err(e) => {
+                if let FreezerState::Frozen = freezer_state {
+                    bail!("freezer not support {}", e);
```

Comment:
Nit: freezer not support**ed**

Reply:
:stuck_out_tongue:
--


<br class="Apple-interchange-newline">